sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_web_pub_sub_service_client.py

```diff
@@ -0,0 +1,209 @@
+        :type accesskey: ~str
+        :keyword ttl: Optional ttl timedelta for the token. Default is 1 hour.
+        :type ttl: ~datetime.timedelta
+        :keyword user: Optional user name (subject) for the token. Default is no user.
+        :type user: ~str
+        :keyword roles: Roles for the token.
+        :type roles: typing.List[str]. Default is no roles.
+        :returns: ~dict containing the web socket endpoint, the token and a url with the generated access token.
+        :rtype: ~dict
+
+
+        Example:
+        >>> build_authentication_token(endpoint='https://contoso.com/api/webpubsub', hub='theHub', key='123')
+        {
+            'baseUrl': 'wss://contoso.com/api/webpubsub/client/hubs/theHub',
+            'token': 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ...',
+            'url': 'wss://contoso.com/api/webpubsub/client/hubs/theHub?access_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ...'
+        }
+        """
+        if 'accesskey' not in kwargs:
+            kwargs = _parse_connection_string(endpoint, **kwargs)
+            endpoint = kwargs.pop('endpoint')
+
+        user = kwargs.pop("user", None)
+        key = kwargs.pop("accesskey")
+        ttl = kwargs.pop("ttl", timedelta(hours=1))
+        roles = kwargs.pop("roles", [])
+        endpoint = endpoint.lower()
+        if not endpoint.startswith("http://") and not endpoint.startswith("https://"):
+            raise ValueError(
+                "Invalid endpoint: '{}' has unknown scheme - expected 'http://' or 'https://'".format(
+                    endpoint
+                )
+            )
+
+        # Ensure endpoint has no trailing slash
+        endpoint = endpoint.rstrip("/")
+
+        # Switch from http(s) to ws(s) scheme
+        client_endpoint = "ws" + endpoint[4:]
+        client_url = "{}/client/hubs/{}".format(client_endpoint, hub)
+        audience = "{}/client/hubs/{}".format(endpoint, hub)
+
+        payload = {
+            "aud": audience,
+            "iat": datetime.now(tz=_UTC),
+            "exp": datetime.now(tz=_UTC) + ttl,
+        }
+        if user:
+            payload["sub"] = user
+        if roles:
+            payload["role"] = roles
+
+        token = six.ensure_str(jwt.encode(payload, key, algorithm="HS256"))
+        return {
+            "baseUrl": client_url,
+            "token": token,
+            "url": "{}?access_token={}".format(client_url, token),
+        }
+
+    @classmethod
+    def from_connection_string(cls, connection_string, **kwargs):
+        # type: (Type[ClientType], str, Any) -> ClientType
+        """Create a new WebPubSubServiceClient from a connection string.
+
+        :param connection_string: Connection string
+        :type connection_string: ~str
+        :rtype: WebPubSubServiceClient
+        """
+        kwargs = _parse_connection_string(connection_string, **kwargs)
+
+        credential = AzureKeyCredential(kwargs.get("accesskey"))
```

Comment:
nit: I think this should be a`pop` instead, since we don't need to perserve `accesskey`